### DISCUSSION TITLE
Actions: Windows "Terminate Batch Job (Y/N)?"

### DISCUSSION THREAD

**Discussion by shellscape** [community/community#26117] [Actions]
<p>It looks like there’s a major bug with Github Actions and the Windows container. We’ve attempted to resolve this with several different iterations on a set of commands - all of which revolved around installing and running PNPM (note that we’re not talking about <em>NPM</em>). Our slate of failures and attempted fixes can be seen here, in all of the failed builds: <a href="https://github.com/rollup/plugins/actions" rel="nofollow noopener">https://github.com/rollup/plugins/actions</a> Note that the builds don’t actually fail, they just stop with some kind of warning state, on the task that is cancelled.</p>
<p>The crux of the problem is that commands are somehow producing a situation in which (so it would appear) some process in the container is attempting to send <code>CTL+C</code> to the shell.</p>
<p><code>Terminate batch job (Y/N)?</code> is immediately followed by <code>##[error]The operation was canceled.</code></p>
<p>From searching the interwebs, this appears to be due to using CMD.exe as the default shell - where Windows requires <code>CTL+C</code> to be sent twice in order to kill a script. So we went so far as to tack on <code>shell: powershell</code> to every command, and that apparently made no difference. This is a pretty big hangup, and is not something that CircleCI nor Travis suffer from (confirmed). This will effectively block the Rollup org from being able to adopt Github Actions.</p>
<p><em>(On another note - Why is the editor for github.community so bad, and why can we not use Markdown?)</em></p>

**Comment by ethomson** [comment, marked as answer]
<p>Digging in a bit more, we discovered that the workflow is actually being cancelled.  The job outputting “Terminate Batch Job (Y/N)?” is part of a matrix.  Matrix builds run in parallel and (by default) the entire matrix will be cancelled if any job fails.</p>
<p>In this case, running a job on a Windows host, you may see this message during the cancellation.  (This is a message that the Windows shell cmd may display during cancellation.)</p>
<p>If you want to turn off the “fail fast” mode you can set:</p>
<pre><code>fail-fast: false
</code></pre>
<p>In your matrix definition.  See the <a href="https://help.github.com/en/articles/workflow-syntax-for-github-actions" rel="nofollow noopener">workflow syntax guide</a> for more information.</p>
